pulsar/consumer_partition.go

```diff
@@ -1564,7 +1578,7 @@ func (pc *partitionConsumer) dispatcher() {
 				pc.log.Debug("skip dispatching messages when seeking")
 			}
 		} else {
-			queueCh = pc.queueCh
+			queueCh = pc.queueCh.outCh
```

Comment:
I find this code a bit convoluted. Can you help me understand why you assign the channel to `queueCh` and then later range over `pc.queueCh` directly if you get a signal to clear the queue?

When is the variable needed and when is not is not very clear. Perhaps a comment might help demistify this?

Reply:
> I wanted to comment also that perhaps using two channels queueInCh and queueOutCh might help make the code more understandable and easier to maintain.

I agree.

> When is the variable needed and when is not is not very clear.

It's only used here:

```go
		case msg, ok := <-queueCh:
			if !ok {
				return
			}

			queueMsg = msg
```

It polls a message from `queueCh`. If the case is hit, then in the next loop, `queueCh` will be nil and it won't try to poll a message from `queueCh`. Instead, `nextMessage` will be the wrapped message so that it will reach here:

```go
		case messageCh <- nextMessage:
			queueMsg = nil
```

I admit it's a bit complicated. The current design comes from https://github.com/apache/pulsar-client-go/pull/86. I'd like not to do some refactoring in this PR so I just changed the actual message channel from `queueCh` to `queueCh.outCh`.

And just as you mentioned, I agree that using two channels will be more clear so I will make some changes.

Reply:
Yeah I agree a refactoring here is beyond the scope of this PR. Perhaps consider adding a comment explaining what you did just here. Might help for future reverse engineering.